lib/warning.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+/**
+ * Class for storing warning details.
+ * @module fs-transform:warning
+ * @author Ryan Sandor Richards
+ */
+module.exports = Warning;
```

Comment:
is there something specific to this repo you want to do here? we could use existing modules like `debug` or `bunyan`